### ISSUE TITLE
`cy.findByText(...).should('not.exist')` returns true for non-selected dropdown option

### ISSUE THREAD

**Issue by vfonic** [testing-library/cypress-testing-library#146]
- `cypress-testing-library` version: 6.0.0
- `node` version: 12.18.2
- `npm` (or `yarn`) version: (yarn) 1.22.4

This issue (in a way) builds on top of @coryhouse's issue #135 `cy.findByText always returns true with should('not.exist')`

I've forked the repo and added the new failing test to `master`:
https://github.com/vfonic/cy-doesnt-exist-bug

Repro steps: clone, `npm i`, `npm run cy`

You can also see the commit diff:
https://github.com/vfonic/cy-doesnt-exist-bug/commit/7cf22ca3189b831010e3c306a0475e8c5da0efc1

Ok, I just made one more commit:
https://github.com/coryhouse/cy-doesnt-exist-bug/compare/master...vfonic:master

(There's "Learn React" `a` tag below that gets correctly selected when the above elements are missing.)

Basically, `findByText` returns the first DOM element by that text.
`should('not.exist')` returns true ("it doesn't exist") if the element is hidden or non-selected dropdown option.

I'm not sure if this is cypress-testing-library issue or cypress/chai issue?
It just sounds misleading, especially for the second case (when `findByText` finds select option)

**Comment by vfonic** [comment]
The solution we found to be working well is:

```js
cy.findAllByText(...).should('not.exist')
```

**Comment by vfonic** [comment]
The difference is between:

```js
cy.findByText(...).should('not.exist')
```

and

```js
cy.findAllByText(...).should('not.exist')
```

Just wanted to point that out. Perhaps the issue is within this library after all?

**Comment by kentcdodds** [comment]
Hmmm... Maybe. I'm afraid I don't completely understand the issue and I haven't had time to dig in. I'll re-open this for anyone who wants to take a crack at it though.

**Comment by davidseow** [comment]
Possibly the reason why this is behaving the way it did. 

&ast;By&ast; (in this case findByText), throws an error when there are more than one element. See: [link](https://github.com/testing-library/dom-testing-library/blob/7f8c100e1139c656ca817c23f223e7c15df26774/src/query-helpers.js#L68)

Error thrown is then captured and an empty jQuery NodeList is returned causing assertion to be true when it should not.
See: [link](https://github.com/testing-library/cypress-testing-library/blob/547389d8605b699687546567914a4598da5a0797/src/index.js#L124) 

At a glance [documentation](https://testing-library.com/docs/dom-testing-library/api-queries#getby) seems to suggest that the first matching node is returned, this is not the case based on above. Would it be clearer if documentation is updated instead?

_From_ 
```
getBy* queries return the first matching node for a query, and throw an error if no elements match or if more than one match is found (use getAllBy instead).
```

_To_
```
getBy* queries return the matching node for a query, and throw an error if no elements match or if more than one match is found (use getAllBy instead).
```

**Comment by kentcdodds** [comment]
Ah, yes that would be a good change to the docs 👍

**Comment by davidseow** [comment]
Thanks, raising a PR now

**Comment by davidseow** [comment]
https://github.com/testing-library/testing-library-docs/pull/728
